### ISSUE TITLE
Request: routing randomisation

### ISSUE THREAD

**Issue by tukkek** [abyssonym/approaching_chaos#3]
Hello, I tried leaving a comment on your blog but I think it didn't get through.

I would love to see some routing randomization for FF1 in addition to everything that is already done in Approaching Chaos. It doesn't have to be anywhere as extensive as [the de-facto NES randomizer](https://github.com/Entroper/FF1Randomizer/) but a basic implementation would be very nice!

Thank you and please let us know if you're interested in adding this at some point. If the basic work is done, I would love to try and port the other more advanced routing features from the NES randomizer myself! I'm completely new to ROM hacking but I think that with the NES implementation as a reference and some basic routing features added to the project I would be able to use them to continue the work!

Thanks again for the cool randomizer. I'm definitely going to give it a spin this weekend :)

**Comment by tukkek** [comment]
I've been studying the FF1 NES randomizer and the Dawn of Souls ROM structure for a bit and i think I get why this isn't an easy task: basically, unlike normal items, spells, shops... quest items are not table-based, which makes them a lot harder to manipulate. Some "quest items" are not items at all and will instead modify the world map - like adding the bridge at the start of the game.

I've been looking into [the relevant NES randomizer code](https://github.com/Entroper/FF1Randomizer/blob/master/FF1Lib/Treasure.cs) and it has some neat ideas, apparently, like copying entire sections of instructions between NPCs to, in practice, swap behavior between them - or placing something like a hidden "chest" where an NPC would otherwise be, letting you take a quest item instead. It also does a fair bit of custom ASM code injection, which I'm not sure if it's viable at all for GBA ROMs.

Sorry if any of this is wrong since I haven't put much time into it but I decided to write it all down anyway in case I (or anyone else) decides to tackle this at some point. The FF1 randomizer definitely proves it's doable but it takes more effort and ROM hacking knowledge than one would assume at first.

**Comment by abyssonym** [comment]
Yeah, that's basically it. The items need not be contiguous... the randomizer can take a list of pointers instead of a table... but because the keys to progression take a variety of forms, it would be awkward and tedious to randomize them.
